typescript/src/request.test.ts

```diff
@@ -22,10 +23,21 @@ test('can pass through data from server', () => {
 });
 
 test('404 creates ApiError', () => {
-  mockFetch(false, {'statusCode': 404}, 404);
+  mockFetch(false, {'statusCode': 404, 'errorCode': 8}, 404);
```

Comment:
I definitely know your intention, but make sure ErrorType.RESOURCE_NOT_FOUND == 8 should be ianother independent testcase or you do not need to test it at all :) if you prefer.

Reply:
> ianother independent testcase

why? it's part of the same concept: if we ask for something that doesn't exist, we get a good error message.

> do not need to test it at all

If i don't test 404 errors, code coverage meter will complain.

If i don't check for the `errorCode` specifically, nothing will complain, but, having a test that the `errorCode` is included seems like a real good idea to me. not including it would be a bug.